Src/java/cql-to-elm/src/main/java/org/cqframework/cql/cql2elm/LibraryBuilder.java

```diff
@@ -2233,17 +2237,23 @@ public Expression resolveIdentifier(String identifier, boolean mustResolve) {
                 if (resolution != null) {
                     Expression contextAccessor = buildProperty(parameterRef, resolution.getName(), resolution.isSearch(), resolution.getType());
                     contextAccessor = applyTargetMap(contextAccessor, resolution.getTargetMap());
-                    return contextAccessor;
+                    matchCollection.add(contextAccessor);
                 }
             }
         }
 
-        if (mustResolve) {
+        if (matchCollection.size() > 0 ){
+            //issue warning that multiple matches occurred:
+            this.reportWarning("Identifier hiding detected: identifier" + (matchCollection.size() > 3 ? "s": "") +" in a broader scope hidden: " + StringUtils.join(matchCollection, ","), matchCollection.get(0));
```

Comment:
The warning should only be returned if the matchCollection has more than one match